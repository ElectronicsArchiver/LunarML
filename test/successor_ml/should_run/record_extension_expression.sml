val x = { a = "a", b = 42 }
fun f x = { c = 0wxFF, ... = x }
val y = f x
val z = { 1 = 100, 2 = 200, ... = y };
print (Int.toString (#1 z) ^ "\n");
print (Int.toString (#2 z) ^ "\n");
print (#a z ^ "\n");
print (Int.toString (#b z) ^ "\n");
print (Word.toString (#c z) ^ "\n");
print (Bool.toString (z = { 1 = 100, 2 = 200, a = "a", b = 42, c = 0wxFF }) ^ "\n");
