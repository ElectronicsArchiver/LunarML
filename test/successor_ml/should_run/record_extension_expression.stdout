100
200
a
42
FF
true
